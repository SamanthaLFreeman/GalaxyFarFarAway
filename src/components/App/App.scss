@font-face {
  font-family: "STARWARS";
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot");
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot?#iefix")
      format("embedded-opentype"),
    url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.woff")
      format("woff"),
    url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.ttf")
      format("truetype"),
    url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.svg#STARWARS")
      format("svg");
}

$star-wars-font-color: #feda4a;

main {
  margin: 0;
  padding: 0;
  background-image: url("../../Images/space\ wallpaper.jpg");
  background-repeat: no-repeat;
  // background-size: cover;
}

h1 {
  color: $star-wars-font-color;
  font-family: STARWARS;
  margin: 0;
  padding: 20px;
  text-align: center;
}
