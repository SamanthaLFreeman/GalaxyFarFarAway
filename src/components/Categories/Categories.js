import React from 'react';
import { NavLink } from 'react-router-dom';
import './Categories.scss';



const Categories = ({checkAvail}) => {
  console.log(checkAvail);
  return (
    <section className='categories'>
      <NavLink to='/people'><button className='people category' onClick={checkAvail}>People</button></NavLink>
      <NavLink to='/planets'><button className='planets category' onClick={checkAvail}>Planets</button></NavLink>
      <NavLink to='/vehicles'><button className='vehicles category' onClick={checkAvail}>Vehicles</button></NavLink>
    </section>
  )
}

export default Categories;