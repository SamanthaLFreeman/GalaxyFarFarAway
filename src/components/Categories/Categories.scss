@import url("https://fonts.googleapis.com/css?family=Jura:500,700&display=swap");

button {
  font-family: "Jura", sans-serif;
  font-weight: bold;
}
