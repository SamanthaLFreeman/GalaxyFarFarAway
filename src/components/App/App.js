import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import Movies from '../Movies/Movies';
import FavR2D2 from '../FavR2D2/FavR2D2';
import Categories from '../Categories/Categories';
import CardContainer from '../CardContainer/CardContainer';
import './App.scss';

class App extends Component {
  constructor() {
    super();
    this.state = {
      film: [],
      people: [],
      planets: []
    };
  }

  componentDidMount() {
    fetch('https://swapi.co/api/people')
      .then(res => res.json())
      .then(data => this.fetchPeople(data.results))
      .then(data => this.fetchSpecies(data))
      .then(data => this.setState({people: data}))

    fetch('https://swapi.co/api/planets/')
      .then(response => response.json())
      .then(data => data.results)
      .then(data => this.createPlanets(data))
      .catch(error => console.log(error));;
    fetch('https://swapi.co/api/vehicles/');

    fetch('https://swapi.co/api/films/')
      .then(response => response.json())
      .then(data => this.setState({ film: data.results[this.getRandomNumber()] }))
      .catch(error => console.log(error));
  }

  fetchPeople = (allPeople) => {
    const promises = allPeople.map(person => {
      return fetch(person.homeworld)
        .then(response => response.json())
        .then(data => ({ 
          name: person.name,
          homeworld: data.name,
          population: data.population,
          species: person.species[0]
        }))
        .catch(error => console.log(error));
    });
    return Promise.all(promises)
  }

  fetchSpecies = (allPeople) => {
    const promises = allPeople.map(person => {
      return fetch(person.species)
        .then(response => response.json())
        .then(data => ({
          ...person,
          species: data.name
        }))
        .catch(error => console.log(error));
    });
    return Promise.all(promises)
  }

  getRandomNumber = () => {
    return Math.floor(Math.random() * Math.floor(7));
  }

  createPlanets(data) {
    let planets = data.map(planet => {
      let planetResidents = [];
        planet.residents.forEach(resident => {
          return fetch(resident)
          .then(response => response.json())
          .then(data => planetResidents.push(data.name));
        });
        return {
          name: planet.name,
          terrain: planet.terrain,
          population: planet.population,
          climate: planet.climate,
          resident: planetResidents
        };
      });
      console.log(planets)
      return planets;
    };

  render() {
    return (
      <main>
        <h1>Galaxy Far Far Away</h1>
        <Movies film={this.state.film}/>
        <FavR2D2 />
        <Categories people={this.state.people} planets={this.state.planets}/>
        <Route exact path='/people' render={() => <CardContainer people={this.state.people}/>} />
        <Route exact path='/planets' render={() => <CardContainer planets={this.state.planets}/>} />
        {/* <Route exact path='/vehicles' render={() => <CardContainer vehicles={this.state.vehicles}/>} /> */}
      </main>
    )
  }
  
}

export default App;
