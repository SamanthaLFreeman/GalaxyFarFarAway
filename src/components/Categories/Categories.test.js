import React from 'react';
import { shallow, mount } from 'enzyme';
import Categories from './Categories';
import App from '../App/App';
import { MemoryRouter } from 'react-router' 

describe('Categories', () => {
  it('should match the snapshot with all the data passed in correctly', () => {
    const wrapper = shallow(<Categories />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Routes', () => {
  it('should route to People', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    )

    expect(wrapper.find(App)).toHaveLength(1)
  })
})