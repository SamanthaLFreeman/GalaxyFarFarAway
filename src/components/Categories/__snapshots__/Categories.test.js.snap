// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="categories"
>
  <NavLink
    to="/people"
  >
    <button
      className="people"
    >
      People
    </button>
  </NavLink>
  <NavLink
    to="/planets"
  >
    <button
      className="planets"
    >
      Planets
    </button>
  </NavLink>
  <NavLink
    to="/vehicles"
  >
    <button
      className="vehicles"
    >
      Vehicles
    </button>
  </NavLink>
</section>
`;
